### PR TITLE
Move Blockly.Block rendered into Blockly.BlockSvg

### PR DESCRIPTION
**Pull request by rachel-fenichel**
@picklesrus

### CODE REVIEW

File: core/block_svg.js [hunk 1]
```diff
@@ -1402,3 +1406,165 @@ Blockly.BlockSvg.prototype.removeDragging = function() {
...
+
+/**
+ * Remove an input from this block.
+ * @param {string} name The name of the input.
+ * @param {boolean=} opt_quiet True to prevent error if input is not present.
+ * @throws {goog.asserts.AssertionError} if the input is not present and
+ *     opt_quiet is not true.
+ */
+Blockly.BlockSvg.prototype.removeInput = function(name, opt_quiet) {
+  Blockly.BlockSvg.superClass_.removeInput.call(this, name, opt_quiet);
```
Comment (picklesrus) [inline core/block_svg.js]: Does this one work right?  Seems like the this.render() call in block.js's removeInput is nested in a loop and some if statements that sometimes won't get hit.  But this doesn't have that logic.  I can't figure out if that's okay or not by just looking at the code.

Comment (rachel-fenichel) [inline core/block_svg.js]: The block.js implementation loops through the inputs looking for the particular relevant input, changes things, rerenders, and returns (short-circuits the rest of the loop).  If it gets through the loop something went wrong.

So in the case where it works (and rendering is on) it calls render exactly once, right before it returns.  This changes it to call render exactly once, right after it returns and before any other code is evaluated.

In the case where something went wrong, there's a failed assert anyway.

Comment (picklesrus) [inline core/block_svg.js]: Ah, yes. The assertion error being thrown was conveniently hiding from me behind the zippy thing.

**Comment by picklesrus** [comment]
LGTM

**Comment by NeilFraser** [comment]
LGTM

**Pull request merged**